core/bcast/recast.go

```diff
@@ -19,15 +21,17 @@ type recastTuple struct {
 // NewRecaster returns a new recaster.
 func NewRecaster() *Recaster {
 	return &Recaster{
-		tuples: make(map[core.PubKey]recastTuple),
+		tuples:       make(map[core.PubKey]recastTuple),
+		uniqueTuples: make(map[core.PubKey]map[eth2p0.Root]core.SignedData),
 	}
 }
 
 // Recaster rebroadcasts core.DutyBuilderRegistration aggregate signatures every epoch.
 type Recaster struct {
-	mu     sync.Mutex
-	tuples map[core.PubKey]recastTuple
-	subs   []func(context.Context, core.Duty, core.PubKey, core.SignedData) error
+	mu           sync.Mutex
+	tuples       map[core.PubKey]recastTuple
+	uniqueTuples map[core.PubKey]map[eth2p0.Root]core.SignedData
```

Comment:
no need for this, can remove, the `tuple.duty.Slot >= duty.Slot` already ensures we store unique tuples.